consumer/consumer.go

```diff
@@ -341,20 +353,35 @@ func (consumer *StateSyncerConsumer) readAndDecodeNextEntry(reader *bufio.Reader
 	if err != nil && (errors.Is(err, io.ErrUnexpectedEOF) || errors.Is(err, io.EOF)) {
 		// If it's an unexpected EOF, log it and return true to signify EOF.
 		glog.V(2).Infof("consumer.readAndDecodeNextEntry: Error reading from state change file: %v", err)
+
 		// Reset the reader to the position before the unexpected EOF.
 		if _, err = file.Seek(currentPos, io.SeekStart); err != nil {
 			return nil, false, errors.Wrapf(err, "consumer.readAndDecodeNextEntry: Error seeking to current position in file")
 		}
 		return nil, true, nil
 	} else if err != nil {
+		if _, err = file.Seek(currentPos, io.SeekStart); err != nil {
```

Comment:
Basically, for every failed state here, since we've already read the length of the encoder from the file and advanced the file reader, we need to revert back to the previous position, so that when we next read from this file, we are currently in the same spot.